src/vscodeUtils.ts

```diff
@@ -28,6 +28,30 @@ export function getActiveTextEditor(): vscode.TextEditor | undefined {
   return vscode.window.visibleTextEditors.find((textEditor) => !!textEditor.viewColumn);
 }
 
+/*
+  Detects whether this is a Terraform file we can perform operations on
+ */
+export function isTerraformFile(document?: vscode.TextDocument): boolean {
+  if (document === undefined) {
+    return false;
+  }
+
+  if (document.isUntitled) {
+    // Untitled files are files which haven't been saved yet, so we don't know if they
+    // are terraform so we return false
+    return false;
+  }
+
+  if (document.fileName.endsWith('tf')) {
+    // For the purposes of this extension, anything with the tf file
+    // extension is a Terraform file
+    return true;
+  }
+
+  // be safe and default to false
+  return false;
+}
+
```

Comment:
Two points/questions:

1) Isn't Untitled document also valid? The user can set the syntax (language ID) explicitly and AFAIK even these documents are normally sent to the LS and LS should be able to process them as virtual documents, even when they don't (yet) exist on the OS filesystem.
2) Is there any way we can avoid this drifting from other logic which tracks what files do we recognize, or ideally avoids the duplication entirely?
   - activation events: https://github.com/hashicorp/vscode-terraform/blob/7dc861c1b6d18d0e5416251fe0528c701b5f6631/package.json#L37-L40
   - association of language IDs with file extensions: https://github.com/hashicorp/vscode-terraform/blob/7dc861c1b6d18d0e5416251fe0528c701b5f6631/package.json#L45-L69

IMHO we should be checking the language ID here at the least (not file extensions) and relatedly check `terraform-vars` in addition to `terraform`.